src/lib/swapper/swappers/OneInchSwapper/utils/fetchOneInchSwap.ts

```diff
@@ -0,0 +1,68 @@
+import { fromAssetId, fromChainId } from '@shapeshiftoss/caip'
+import { getConfig } from 'config'
+import type { Asset } from 'lib/asset-service'
+import { bnOrZero } from 'lib/bignumber/bignumber'
+import { convertBasisPointsToPercentage } from 'state/zustand/swapperStore/utils'
+
+import { getTreasuryAddressFromChainId } from '../../utils/helpers/helpers'
+import { DEFAULT_SLIPPAGE } from './constants'
+import { oneInchService } from './oneInchService'
+import type { OneInchSwapApiInput, OneInchSwapResponse } from './types'
+
+export type FetchOneInchSwapInput = {
+  affiliateBps: string
+  buyAsset: Asset
+  receiveAddress: string
+  sellAmountBeforeFeesCryptoBaseUnit: string
+  sellAsset: Asset
+  slippage?: string
+}
+
+export const fetchOneInchSwap = async ({
+  affiliateBps,
+  buyAsset,
+  receiveAddress,
+  sellAmountBeforeFeesCryptoBaseUnit,
+  sellAsset,
+  slippage,
+}: FetchOneInchSwapInput) => {
+  const apiUrl = getConfig().REACT_APP_ONE_INCH_API_URL
+
+  /**
+   * limit of price slippage you are willing to accept in percentage,
+   * may be set with decimals. &slippage=0.5 means 0.5% slippage is acceptable.
+   * Low values increase chances that transaction will fail,
+   * high values increase chances of front running. Set values in the range from 0 to 50
+   */
+  const slippagePercentage = (slippage ? bnOrZero(slippage) : bnOrZero(DEFAULT_SLIPPAGE))
```

Comment:
In addition to the great comment above, do we want to also rename this to something like `maximumSlippagePercentage` or `requestedSlippagePercentage`?
The reason for this being it easier to grep for *input* / request slippage (i.e what we have currently) and *quote* slippage (which we don't have currently, but may)

Reply:
Also will fix in follow-up

Reply:
7dd8621de8c7a1329574050bf20d2e2a702163ec